### PR TITLE
Implement due dates related methods for Task and TaskMut

### LINKED ISSUE #2879: [taskchampion] Implement due dates
See https://github.com/taskchampion/taskchampion/issues/82

In the context of Taskwarrior, this only entails adding due-date-related methods to Task and TaskMut.

### PR DESCRIPTION
**Pull request by Treenhan**
#### Description

Implement due dates related methods for Task and TaskMut

### Additional information
All tests pass

Fixes https://github.com/GothenburgBitFactory/taskwarrior/issues/2879

### CODE REVIEW

File: taskchampion/taskchampion/src/task/task.rs [hunk 1]
```diff
@@ -458,6 +464,14 @@ impl<'r> TaskMut<'r> {
         self.set_string(format!("annotation_{}", entry.timestamp()), None)
     }
 
+    pub fn add_due(&mut self, due: Timestamp) -> Result<()> {
+        self.set_timestamp(Prop::Due.as_ref(), Some(due))
+    }
+
+    pub fn remove_due(&mut self) -> Result<()> {
+        self.set_string(Prop::Due.as_ref(), None)
+    }
```
Comment (djmitche) [inline taskchampion/taskchampion/src/task/task.rs]: To parallel `set_entry` and `set_wait`, let's have this take an `Option<Timestamp>` and set or remove the value depending on whether that's Some or None.

File: taskchampion/taskchampion/src/task/task.rs [hunk 2]
```diff
@@ -745,6 +759,53 @@ mod test {
...
+            vec![(String::from("due"), format!("{}", test_time.timestamp()))]
+                .drain(..)
+                .collect(),
+            dm(),
+        );
+        assert_eq!(task.get_due(), Some(test_time))
+    }
+
+    #[test]
+    fn test_get_invalid_due() {
```
Comment (djmitche) [inline taskchampion/taskchampion/src/task/task.rs:772]: 👍🏽

**Review by djmitche — COMMENTED** [review]
Thanks -- looking good! Once this is done, the next PR will be to reflect it in `taskchampion/lib`

File: taskchampion/taskchampion/src/task/task.rs (hunk 1 above)
Comment (Treenhan) [reply taskchampion/taskchampion/src/task/task.rs]: great suggestion! code updated

**Review by djmitche — APPROVED** [review]
Looks good! I'll get this merged, and the next bit is to add functionality in `taskchampion/lib` to match this.

**Pull request merged**